coderd/workspaceapps/db.go

```diff
@@ -447,16 +447,20 @@ func (p *DBTokenProvider) auditInitRequest(ctx context.Context, w http.ResponseW
 			slog.F("status_code", statusCode),
 		)
 
-		var startedAt time.Time
+		var (
+			refreshID       = uuid.New()
+			upsertRefreshID uuid.UUID
+		)
```

Comment:
I feel this is potentially error-prone.
Apparently you can use `(xmax = 0) AS inserted` to know if we did an insert instead of an update ([link](https://stackoverflow.com/questions/39058213/differentiate-inserted-and-updated-rows-in-upsert-using-system-columns)). Would this be a better option?

Reply:
Which error risk did you identify?

The `xmax` behavior can be used to differentiate _inserts_ and _updates_, but there are two types of _updates_ in this case which it doesn't help with. Btw, `xmax` behavior is something of an implementation detail, not sure I'd want to depend on it in production code.

Reply:
It requires you to know beforehand that the ID you put in is not necessarily the ID you get out and that you may want to check that what you put in is what you get out. So more of a potential sharp edge for the unwary.

Reply:
Thanks for explaining the problem, made this a simple fix a00d197.